libs/wingsdk/src/testing/simulator.ts

```diff
@@ -231,6 +246,66 @@ export class Simulator {
     this._running = true;
   }
 
+  /**
+   * Start a server that allows any resource to be accessed via HTTP.
+   */
+  public async startServer(): Promise<void> {
+    const app = express();
+    app.use(express.json());
+
+    app.post("/v1/call", async (req, res, next) => {
+      try {
+        let request: SimulatorServerRequest = req.body;
+        // assumption: Wing APIs don't distinguish between null and undefined, so we can swap them
+        request = JSON.parse(JSON.stringify(request), (_key, value) => {
+          return value === null ? undefined : value;
+        });
+
+        const { handle, method, args } = request;
+        const resource = this._handles.find(handle);
+
+        try {
+          const result = await (resource as any)[method](...args);
+          // assumption: Wing APIs are only returning JSON-serializable values
```

Comment:
This needs to be enforced by the compiler for user defined inflight APIs of preflight classes and also in our sdk using the api checker. 

I am ok if this is not part of this PR but we should fast follow because it will easily break

Reply:
Opened https://github.com/winglang/wing/issues/4510 👍